### PR TITLE
Create new conda environment in yml with Obspy 1.2.2 and PySide2 and fix compatible issues

### PR DESCRIPTION
**Pull request by damhuonglan**
Obspy 1.2.2 was installed for PIC's conda environment to be similar to the one in DMC. However some GUI programs in PH5 require PyQt4 and PySide which cause conflicts. We decided to install PySide2 instead.
This PR is to modify the codes to work with PySide.
The PR #454 that make seg2toph5 compatible with both Obspy 1.1.1 and 1.2.2 has been merged to this PR to test pforma.
The PR #457 that updated environment.yml with Obspy 1.2.2 and PySide2 has been merged to this PR which make this PR combining all needed to change conda environment.

### Checklist
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] All tests pass.
- [ ] Any new or changed features have are documented.
- [ ] Changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Review by hrotman-pic — APPROVED** [review]
I have tested pforma with a small (2 files) Fairfield dataset, and used experiment_t_gen and noven to create appropriate kef files for that small dataset.  The small test passes ph5_validate and outputs stationXML.

I also tested pforma with two larger datasets: a ~200 GB (30 files) Fairfield, and ~55 GB (~380 files) Texan.  Both loaded normally, passed validate, and output XML and data.

I also used noven to create some kefs, including checking a csv that had a problem in it, and noven behaves as expected.

All utilities appear to work the same as they did previously.  Thank you.

**Comment by damhuonglan** [comment]
Thanks @hrotman-pic. But since PH52 always update with the newest PH5 from git, I will not merge this PR until PH52 is update with obspy 1.2.2 and PySide2.

File: ph5/utilities/noven.py [hunk 1]
```diff
@@ -13,18 +13,15 @@
...
+    from PySide2 import QtWidgets, QtCore
 except Exception:
-    msg = ("\n\nNo module named PyQt4. "
-           "Please install PyQt4 first, it is needed for noven. "
-           "\n\n"
-           "If using Anaconda run 'conda install pyqt=4'"
-           "For pip users, PyQt4 installation instructions are available at "
-           "http://pyqt.sourceforge.net/Docs/PyQt4/installation.html.")
+    msg = ("\n\nNo module named PySide2. "
+           "Please environment_gui.yml to install conda environment"
```
Comment (MaevaP) [inline ph5/utilities/noven.py:23]: "Please use environment_gui.yml to install all required dependencies..." or something similar to that.

File: ph5/utilities/novitiate.py [hunk 2]
```diff
@@ -18,12 +18,15 @@
...
-PROG_VERSION = __version__ = "2019.14"
+PROG_VERSION = __version__ = "2021.84"
 LOGGER = logging.getLogger(__name__)
 try:
-    from PyQt4 import QtGui, QtCore, Qt
+    from PySide2 import QtWidgets, QtCore, Qt
 except Exception:
-    LOGGER.error("PyQt4 must be installed for this to run")
+    msg = ("\n\nNo module named PySide2. "
+           "Please environment_gui.yml to install conda environment"
```
Comment (MaevaP) [inline ph5/utilities/novitiate.py:27]: "Please use environment_gui.yml to install all required dependencies..." or something similar to that.

File: ph5/utilities/pformagui.py [hunk 3]
```diff
@@ -10,23 +10,20 @@
...
+    from PySide2 import QtCore, QtWidgets
 except Exception:
-    msg = ("\n\nNo module named PySide. "
-           "Please install PySide first, it is needed for pforma. "
-           "\n\n"
-           "If using Anaconda run 'conda install PySide'"
-           "For pip users, PySide installation instructions are available at "
-           "https://pypi.org/project/PySide/#installation.")
+    msg = ("\n\nNo module named PySide2. "
+           "Please environment_gui.yml to install conda environment"
```
Comment (MaevaP) [inline ph5/utilities/pformagui.py:19]: "Please use environment_gui.yml to install all required dependencies..." or something similar to that.

**Review by MaevaP — APPROVED** [review]
Looks good to me. Just have a small suggestion for the message outputted when an `Exception` error is raised when trying to import some modules from the `PySide2` library.

**Comment by hrotman-pic** [comment]
> Thanks @hrotman-pic. But since PH52 always update with the newest PH5 from git, I will not merge this PR until PH52 is update with obspy 1.2.2 and PySide2.

I understand.

A question about novitiate, since it is no longer in use: should it be removed, or given some note to clearly label it as not needing to be kept up to date?

**Comment by damhuonglan** [comment]
> A question about novitiate, since it is no longer in use: should it be removed, or given some note to clearly label it as not needing to be kept up to date?

I will ask this question in next PH5 meeting.
Thanks about mentioning, I don't want to work on something for no use.

**Comment by timronan** [comment]
Is `novitiate` related to this issue? Would this be extraneous to this PR? It may be worth opening a new issue for `novitiate` but I am not sure.

**Comment by hrotman-pic** [comment]
@damhuonglan you are welcome, I mentioned it for that reason.  #348 seems relevant, we may want to consider re-opening.

@timronan novitiate was modified under this PR even though it has fallen out of use.  I agree that how to label it and similar utilities probably needs a different issue and/or PR.

**Comment by damhuonglan** [comment]
I think I should return novitiate.py back to the way it is in master now. We may have another PR to remove it or note it as no need to be updated.

**Comment by timronan** [comment]
@hrotman-pic I agree and think we should reopen issue #348. Its good to at least think about remove the unused code as we move forwards. 

Removing code could be tricky so we want to do it in induvidual PRs to makes sure that we have a rollback strategy and not over commit ourselves.

**Review by hrotman-pic — APPROVED** [review]
Pforma ran without error on SEG-2 data on ph51, so I think this is working as intended.  Thank you.

**Pull request merged**